### PR TITLE
Change training for foot down detector to be more robust and train directly from NBS files

### PR DESCRIPTION
**Pull request by TrentHouliston**

### CODE REVIEW

File: module/platform/darwin/SensorFilter/src/VirtualLoadSensor.h [hunk 1]
```diff
@@ -20,52 +20,141 @@
...
+                : noise_factor(0.0f), current_noise(0.0f), certainty_threshold(0.0f), uncertainty_threshold(0.0f) {}
+            VirtualLoadSensor(const ::extension::Configuration& config) {
+                // Bayes settings
+                noise_factor          = config["filter"]["noise_factor"].as<Scalar>();
+                certainty_threshold   = config["filter"]["certainty_threshold"].as<Scalar>();
+                uncertainty_threshold = config["filter"]["uncertainty_threshold"].as<Scalar>();
+
+                // Add the servos
+                for (const auto& field : config["network"]["input"]["servos"].config) {
+                    servos.emplace_back("R_" + field.as<std::string>());
```
Comment (TrentHouliston) [inline module/platform/darwin/SensorFilter/src/VirtualLoadSensor.h]: This should be L, R not R, L

File: module/platform/darwin/SensorFilter/src/VirtualLoadSensor.h [hunk 2]
```diff
@@ -20,52 +20,156 @@
...
+                    else if (field.as<std::string>() == "VELOCITY") {
+                        fields.emplace_back(VELOCITY);
+                    }
+                }
+
+                accelerometer = config["network"]["input"]["accelerometer"].as<bool>();
+                gyroscope     = config["network"]["input"]["gyroscope"].as<bool>();
+
+
+                for (const auto& layer : config["network"]["layers"].config) {
```
Comment (Bidski) [inline module/platform/darwin/SensorFilter/src/VirtualLoadSensor.h]: This should use a YAML Expression

File: module/platform/darwin/SensorFilter/src/VirtualLoadSensor.h [hunk 3]
```diff
@@ -20,52 +20,156 @@
...
+                        }
+                        bias[col] = layer["biases"][col].as<Scalar>();
+                    }
+
+                    layers.emplace_back(weights, bias);
+                }
+            }
+
+            std::array<bool, 2> updateFeet(const ::message::input::Sensors& sensors) {
+
```
Comment (Bidski) [inline module/platform/darwin/SensorFilter/src/VirtualLoadSensor.h]: So much whitespace!

File: tools/footdown.py [hunk 4]
```diff
@@ -0,0 +1,301 @@
+#!/usr/bin/env python3
+
+import os
+import tensorflow as tf
```
Comment (Bidski) [inline tools/footdown.py]: Wrap potentially contentious imports in a try...except

Comment (TrentHouliston) [reply tools/footdown.py]: Should do this for all of them and provide/not provide a b command depending on this (another PR)

**Pull request merged**